x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/swimlane/swimlane_connectors.tsx

```diff
@@ -0,0 +1,385 @@
+  //     toastNotifications: toasts,
+  //     actionConnector: this,
+  //     id: appId,
+  //   });
+  // };
+  const abortCtrl = useRef(new AbortController());
+
+  const onSearchChange = useCallback(
+    (searchValue) => {
+      setLoading(true);
+      setOptions([]);
+
+      const didCancel = false;
+
+      const fetchData = async () => {
+        if (!apiToken || !apiUrl || !appId) {
+          setLoading(false);
+          return;
+        }
+
+        abortCtrl.current = new AbortController();
+        setLoading(true);
+
+        try {
+          const res = await getApplication({
+            http,
+            signal: abortCtrl.current.signal,
+            connectorId: action.id,
+            id: appId,
+          });
+          if (!didCancel) {
+            setLoading(false);
+            const application = res.data?.application;
+            if (application?.fields) {
+              const applicationFields = application.fields;
+              const fieldMap = applicationFields.map((f: { id: string; key: string }) => ({
+                value: f.id,
+                label: f.key,
+              }));
+              setOptions(fieldMap);
+            }
+            if (res.status && res.status === 'error') {
+              toasts.addDanger({
+                title: i18n.SW_GET_APPLICATION_API_ERROR(appId),
+                text: `${res.serviceMessage ?? res.message}`,
+              });
+            }
+          }
+        } catch (error) {
+          if (!didCancel) {
+            setLoading(false);
+            toasts.addDanger({
+              title: i18n.SW_GET_APPLICATION_API_ERROR(appId),
+              text: error.message,
+            });
+          }
+        }
+      };
+
+      abortCtrl.current.abort();
+      fetchData();
+    },
+    [toasts, action, appId, apiToken, http, apiUrl]
+  );
+
+  useEffect(() => {
+    // Simulate initial load.
+    onSearchChange('');
+  }, [onSearchChange]);
+
+  return (
+    <Fragment>
+      <EuiFormRow id="apiUrl" fullWidth label={i18n.SW_API_URL_TEXT_FIELD_LABEL}>
+        <EuiFieldText
+          fullWidth
+          name="apiUrl"
+          value={apiUrl}
+          readOnly={readOnly}
+          data-test-subj="swimlaneApiUrlInput"
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+            editActionConfig('apiUrl', e.target.value);
+          }}
+          onBlur={() => {
+            if (!apiUrl) {
+              editActionConfig('apiUrl', '');
+            }
+          }}
+        />
+      </EuiFormRow>
+      <EuiFormRow id="appId" fullWidth label={i18n.SW_APP_ID_TEXT_FIELD_LABEL}>
+        <EuiFieldText
+          fullWidth
+          name="appId"
+          value={appId || ''}
+          readOnly={readOnly}
+          data-test-subj="swimlaneAppIdInput"
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+            editActionConfig('appId', e.target.value);
+          }}
+          onBlur={() => {
+            if (!apiUrl) {
```

Comment:
I think that's bad CTR-C CTR:_V ;)
```suggestion
            if (!apiId) {
```